src/mlpack/tests/main_tests/kmeans_test.cpp

```diff
@@ -398,17 +390,23 @@ BOOST_AUTO_TEST_CASE(AlgorithmsSimilarTest)
   dualCoverTreeOutput = std::move(CLI::GetParam<arma::mat>("output"));
   dualCoverTreeCentroid = std::move(CLI::GetParam<arma::mat>("centroid"));
 
-  // Check That all the algorithms yield the same clusters
-  CheckMatrices(naiveOutput, elkanOutput);
-  CheckMatrices(elkanOutput, hamerlyOutput);
-  CheckMatrices(hamerlyOutput, dualTreeOutput);
-  CheckMatrices(dualTreeOutput, dualCoverTreeOutput);
-
-  CheckMatrices(naiveCentroid, elkanCentroid);
-  CheckMatrices(elkanCentroid, hamerlyCentroid);
-  CheckMatrices(hamerlyCentroid, dualTreeCentroid);
-  CheckMatrices(dualTreeCentroid, dualCoverTreeCentroid);
+  for (size_t i = 0; i < inputData.n_cols; ++i)
+    BOOST_REQUIRE_EQUAL(naiveOutput(0, i), elkanOutput(0, i));
+  for (size_t i = 0; i < inputData.n_cols; ++i)
+    BOOST_REQUIRE_EQUAL(naiveOutput(0, i), hamerlyOutput(0, i));
+  for (size_t i = 0; i < inputData.n_cols; ++i)
+    BOOST_REQUIRE_EQUAL(naiveOutput(0, i), dualTreeOutput(0, i));
+  for (size_t i = 0; i < inputData.n_cols; ++i)
+    BOOST_REQUIRE_EQUAL(naiveOutput(0, i), dualCoverTreeOutput(0, i));
+
+  for (size_t i = 0; i < initCentroid.n_elem; ++i)
+    BOOST_REQUIRE_CLOSE(naiveCentroid(0, i), elkanCentroid(0, i), 1e-5);
+  for (size_t i = 0; i < initCentroid.n_elem; ++i)
+    BOOST_REQUIRE_CLOSE(naiveCentroid(0, i), hamerlyCentroid(0, i), 1e-5);
+  for (size_t i = 0; i < initCentroid.n_elem; ++i)
+    BOOST_REQUIRE_CLOSE(naiveCentroid(0, i), dualTreeCentroid(0, i), 1e-5);
+  for (size_t i = 0; i < initCentroid.n_elem; ++i)
+    BOOST_REQUIRE_CLOSE(naiveCentroid(0, i), dualCoverTreeCentroid(0, i), 1e-5);
```

Comment:
I'm confused by this change; what was wrong with `CheckMatrices()`?

Reply:
Don't get confused. Its just that while handling the issue I adopted this style for checking to have more control over the tolerance but now I 've reverted it back to `CheckMatrices()`